metrics.go

```diff
@@ -15,55 +15,105 @@ type CertificateDesc struct {
 }
 
 type MetricsCollector struct {
-	certsMetric *prometheus.Desc
+	certsMetric  *prometheus.Desc
+	errorsMetric *prometheus.Desc
 
-	role           string
-	updates        chan VaultEvent
-	resourceEvents map[string]VaultEvent
-	resourcesMutex sync.RWMutex
+	role string
+
+	resourcesUpdates chan VaultEvent
+	resourceEvents   map[string]VaultEvent
+
+	errorUpdates  chan string
+	errorCounters map[string]int
+
+	metricsMutex sync.RWMutex
+}
+
+var errorUpdates chan string
+var errorUpdatesMutex sync.Mutex
+
+func setErrorUpdatesChan(ch chan string) {
```

Comment:
Perhaps instead of having these `{set,get}ErrorUpdatesChan` functions, we could just make the metrics collector a top level variable. 

So at the top of this file we could have 
` var metrics *MetricsCollector`
and then this would be initialised  in `RegisterMetricsCollector`. Then we would just need to have one `errorUpdates`, which would be inside of the `MetricsCollector` struct. When we want to increment a metric from elsewhere in the code, we can use `metrics.IncremenetErrorCounters(..)`.

What do you think? I think this would mean there are less top level things to keep track off, and it makes the thee `errorUpdates` code more symmetric to the `resourcesUpdates` code.

Reply:
Makes sense 👍 
I'll make the whole `MetricsCollector` structure a global variable